### PR TITLE
Numerical derivatives must apply ties after parameter shift.

### LINKED ISSUE #22004: Results table in muon analysis
If you do a multiple fit in muon analysis with some global values and some fixed values, then the results table will have global values printed for every workspace (the values will also be slightly different).

### PR DESCRIPTION
**Pull request by mantid-roman**
Call `applyTies()` after a shift is added to a parameter when calculating the numerical derivatives.

**To test:**

Instructions are in the issue description.

Fixes #22004

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by AnthonyLim23 — CHANGES_REQUESTED** [review]
We should add something to the release notes. I have also just noticed that the errors are not copied over for the ties (is this easy to fix)? 

The code change looks good and it works

**Comment by mantid-roman** [comment]
@AnthonyLim23 Can you suggest what to say in the release notes? We cannot say that all issues with ties have been fixed.

**Comment by AnthonyLim23** [comment]
We could just say that tied parameters are updated correctly in multiple fits (or do you think this is too vague)

**Comment by mantid-roman** [comment]
You can still set ties that won't be updated correctly

**Comment by AnthonyLim23** [comment]
ah. Ok we will leave it then. Will the errors being copied be much work?

**Comment by mantid-roman** [comment]
Sorry, I didn't understand the question

**Comment by AnthonyLim23** [comment]
If you look at the screenshots from the issue you will see that the errors are not copied to the other workspaces. How much work would that be to fix?

**Comment by mantid-roman** [comment]
Do they need to be copied?

**Comment by AnthonyLim23** [comment]
just thought it could be confusing

**Comment by mantid-roman** [comment]
I guess it depends on the point of view. But if it's what the scientists want it shouldn't be too hard to add.

File: Testing/SystemTests/tests/analysis/reference/II.IRISFuryFitMulti.nxs.md5 [hunk 1]
```diff
@@ -1 +1 @@
-ce0d536337942b5662b8ecd107a5c71f
+d3c125646057bc0d1291c3286ddfab82
```
Comment (AnthonyLim23) [inline Testing/SystemTests/tests/analysis/reference/II.IRISFuryFitMulti.nxs.md5]: what is the difference between these two files? Is it just the parameters being copied correctly?

Comment (mantid-roman) [reply Testing/SystemTests/tests/analysis/reference/II.IRISFuryFitMulti.nxs.md5]: There was one number different

Comment (AnthonyLim23) [reply Testing/SystemTests/tests/analysis/reference/II.IRISFuryFitMulti.nxs.md5]: was it by much?

**Comment by AnthonyLim23** [comment]
for the release notes could we say:

Fix for a bug in calculating numerical derivatives by applying ties correctly. 

?

File: Testing/SystemTests/tests/analysis/reference/II.IRISFuryFitMulti.nxs.md5 (hunk 1 above)
Comment (mantid-roman) [reply Testing/SystemTests/tests/analysis/reference/II.IRISFuryFitMulti.nxs.md5]: Less than 0.1%

**Comment by mantid-roman** [comment]
This is more accurate

**Review by AnthonyLim23 — APPROVED** [review]

**Comment by NickDraper** [comment]
builds cancelled to free up Q, final change was release notes only

**Comment by AnthonyLim23** [comment]
but the last few tests never finished on the previous build

**Pull request merged**